packages/preferences/src/browser/workspace-preference-provider.ts

```diff
@@ -46,14 +46,16 @@ export class WorkspacePreferenceProvider extends PreferenceProvider {
                 this._ready.resolve();
             }
         });
-        this.toDispose.push(this.toDisposeOnEnsureDelegateUpToDate);
         this.workspaceService.onWorkspaceLocationChanged(() => this.ensureDelegateUpToDate());
         this.workspaceService.onWorkspaceChanged(() => this.ensureDelegateUpToDate());
     }
 
     getConfigUri(resourceUri: string | undefined = this.ensureResourceUri(), sectionName?: string): URI | undefined {
-        const delegate = this.delegate;
-        return delegate && delegate.getConfigUri(resourceUri, sectionName);
+        return this.delegate?.getConfigUri(resourceUri, sectionName);
+    }
+
+    getContainingConfigUri(resourceUri: string | undefined = this.ensureResourceUri(), sectionName?: string): URI | undefined {
+        return this.delegate?.getContainingConfigUri?.(resourceUri, sectionName);
```

Comment:
@vince-fugnitto, adding this method to the `WorkspacePreferenceProvider` should solve the problem you were seeing where single-root, unsaved workspaces had trouble opening the JSON.